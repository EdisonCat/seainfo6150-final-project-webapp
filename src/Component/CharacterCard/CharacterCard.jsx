import React from "react";
import styles from "./CharacterCard.module.css";
import { useHistory } from "react-router-dom";
export default function CharacterCard(props) {
	const history = useHistory();
	const handleClick = () => {
		history.push({ pathname: "/characters/" + props.data.id });
	};
	return (
		<div className={styles.container}>
			<div>
				<h1>{props.data.name}</h1>
			</div>
			<img
				className={styles.thumbnail}
				src={props.data.thumbnail.path + ".jpg"}
				alt="thumbnail"
			/>
			<button onClick={() => handleClick()}>More Info</button>
		</div>
	);
}
