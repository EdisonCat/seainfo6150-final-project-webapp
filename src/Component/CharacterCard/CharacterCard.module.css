@import url('https://fonts.googleapis.com/css2?family=Work+Sans:wght@300&display=swap');
.container {
	/* border: white solid 1px; */
	display: flex;
	flex-direction: column;
	align-items: center;
	color: white;
	margin: 10px;
	padding: 20px;
	height: 500px;
	justify-content: space-around;
	font-family: 'Work Sans', sans-serif;


	border-radius: 53px;
	background: #333333;
	box-shadow: inset 25px 25px 50px #141414, inset -25px -25px 50px #525252;
    transition: 0.5s;
}
.container:hover {
	border-radius: 53px;
	background: linear-gradient(145deg, #373737, #2e2e2e);
	box-shadow: 25px 25px 50px #141414, -25px -25px 50px #525252;
}
.container button {
	background: white;
	transition: 0.5s;
	width: 100px;
}
.container button:hover {
	background: red;
	transition: 0.5s;
	width: 120px;
	color: white;
}
.thumbnail {
	max-width: 300px;
	max-height: 280px;
	width: 100%;
	height: auto;
}
h1 {
	font-size: 20px;
}
button {
	width: 100px;
	border: none;
	border-radius: 30px;
	padding: 5px;
	margin: 10px;
	transition: 0.5s;
	cursor: pointer;
    outline: none;
}
button:hover {
	width: 120px;
	color: white;
	background: red;
}
