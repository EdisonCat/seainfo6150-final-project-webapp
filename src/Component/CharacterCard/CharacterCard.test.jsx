import React from "react";
import CharacterCard from "./CharacterCard";
it("rendered CharacterCard correctly", () => {
    const data = {
        id: 0,
        name: "",
        thumbnail: {
            path: "",
            extension: ""
        }
    }
	// eslint-disable-next-line no-undef
    const { container } = render(<CharacterCard data={data}/>);
    expect(container).toMatchSnapshot();
});