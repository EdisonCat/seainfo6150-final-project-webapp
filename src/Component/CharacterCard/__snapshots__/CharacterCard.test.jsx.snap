// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered CharacterCard correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <h1 />
    </div>
    <img
      alt="thumbnail"
      class="thumbnail"
      src=".jpg"
    />
    <button>
      More Info
    </button>
  </div>
</div>
`;
